src/wallet/coinselection.cpp

```diff
@@ -112,38 +111,39 @@ bool SelectCoinsBnB(std::vector<CInputCoin>& utxo_pool, const CAmount& target_va
         // Backtracking, moving backwards
         if (backtrack) {
             // Walk backwards to find the last included UTXO that still needs to have its omission branch traversed.
-            while (!curr_selection.empty() && !curr_selection.back()) {
-                curr_selection.pop_back();
```

Comment:
As an extra belt-and-suspenders check, walking backwards should reset each selection choice to the default state of false.